connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/RestServer.java

```diff
@@ -419,7 +419,7 @@ else if (listeners.contains(String.format("%s://", PROTOCOL_HTTPS)))
     ServerConnector findConnector(String protocol) {
         for (Connector connector : jettyServer.getConnectors()) {
             String connectorName = connector.getName();
-            if (connectorName.startsWith(protocol) && !ADMIN_SERVER_CONNECTOR_NAME.equals(connectorName))
+            if (connectorName.startsWith(protocol + "_") && !ADMIN_SERVER_CONNECTOR_NAME.equals(connectorName))
```

Comment:
I see how the names are by-convention of the form `{protocol}_{host}{port}` and that this technically works. Seems like at a minimum we need a comment (or better yet JavaDoc) that describes why this works and the assumptions it makes.

Reply:
👍 for the comment, so we won't have to rediscover this convention in the future.

Reply:
Ack, will address